### DISCUSSION TITLE
XIRR for capital deployed

### DISCUSSION THREAD

**Discussion by nareshtedla** [kernc/backtesting.py#347] [Q&A]
Hi,

I have been doing back testing for a strategy for a period of 3 years of data. My returns are as below based on the stats:
Return [%]                         114.659196
Buy & Hold Return [%]              209.190101
Return (Ann.) [%]                   27.910586
Trades                                  379
Avg. Trade Duration           1 days 04:49:00

Return (Ann.) [%] assumes that my money is sitting idle when there is no trade. However, in real world situation, money is deployed into other trades. Is there a easy way to calculate the XIRR only for the period of time money is deployed instead of the whole duration?

Also, XIRR for Buy & Hold duration?

Any help in this regard is greatly appreciated. Thanks.

**Comment by kernc** [comment]
> Return (Ann.) [%] assumes that my money is sitting idle when there is no trade. However, in real world situation, money is deployed into other trades.

You could look into and extrapolate `stats['Exposure Time [%]']`. If exposure time is, say, 10%, you could assume the real returns to be 10-times greater if all the money were allocated all the time in strategies of similar performance.

**Comment by nareshtedla** [comment]
This actually helps to an extent. Thanks for quick turnaround.

**Reply by kernc** [reply]
But, now reading what I wrote, the linear 10x doesn't take into account compounding, making it a pessimistic (i.e. underestimating) measure. :sweat_smile:

**Reply by nareshtedla** [reply]
Agreed. Compounding can happen either way :-)
